adala/cli.py

```diff
@@ -0,0 +1,91 @@
+import argparse
+import sys
+import pandas as pd
+from rich import print
+from adala.utils.gsheet import read_gsheet, write_gsheet
+from adala.agents.base import create_agent_from_file
+
+
+def run_gsheet(agent, input_file, worksheet):
+    print(f"Reading input from {input_file}...")
+    input = read_gsheet(input_file, worksheet)
+    print(f"Running agent on {input.shape[0]} rows...")
+    predictions = agent.run(input)
+    print(f"Writing predictions to {input_file} ({worksheet})...")
+    write_gsheet(predictions, input_file, worksheet)
+
+
+def _run_df(agent, df, output_file):
+    print(f"Running agent on {df.shape[0]} rows...")
+    predictions = agent.run(df)
+    if output_file is None:
+        print("Result:")
+        print(predictions)
+    else:
+        predictions.to_csv(output_file, index=False)
+
+
+def run_csv(agent, input_file, output_file):
+    print(f"Reading input from {input_file}...")
+    input = pd.read_csv(input_file)
+    _run_df(agent, input, output_file)
+
+
+def run_dataset(agent, input_file, dataset_config, dataset_split, output_file):
+    from datasets import load_dataset
+
+    ds = load_dataset(input_file, dataset_config, split=dataset_split)
+    input = pd.DataFrame(ds)
+    _run_df(agent, input, output_file)
+
+
+def run(args):
+    print(f"Creating agent from {args.workflow}...")
+    agent = create_agent_from_file(args.workflow)
+    print(agent)
+    if args.file_type == "gsheet":
+        run_gsheet(agent, args.input, args.worksheet)
+    elif args.file_type == "csv":
+        run_csv(agent, args.input, args.output)
+    elif args.file_type == "dataset":
+        run_dataset(
+            agent, args.input, args.dataset_config, args.dataset_split, args.output
+        )
+    else:
+        raise ValueError(f"Unknown file type {args.file_type}")
+    print("Done!")
+
+
+def parse_arguments(args):
```

Comment:
I think CLI should be a separate addition, i.e. PR, and designed properly from the ground up so that we don't end up introducing breaking changes.

Reply:
ok, we can hide it from this PR, anything in mind for the specific CLI design?